frontend/src/components/UI/Alert/VanishingAlert.jsx

```diff
@@ -0,0 +1,57 @@
+import PropTypes from "prop-types";
+import { useEffect } from "react";
+import SimpleAlert from "./SimpleAlert";
+
+/**
+ * An alert component that dismisses itself after a set amount of time.
+ * @param {Object} props - The props object.
+ * @param {React.ReactNode} [props.children] - The child elements to render inside the alert. - optional
+ * @param {string} props.heading - The heading text for the alert.
+ * @param {string} props.message - The message text for the alert.
+ * @param {string} props.type - The type of alert to display.
+ * @param {boolean} [props.isAlertVisible] - Whether the alert is visible.
+ * @param {Function} [props.setIsAlertVisible] - The function to set the alert visibility.
+ * @param {number} [props.timeout] - The amount of time before the alert disappears in milliseconds. Default is 2000ms
+ * @returns {JSX.Element | null} - The rendered alert component.
```

Comment:
🤔 not sure this component will ever return null?

Reply:
It won't, removed this.